tests/integration/test_charm.py

```diff
@@ -304,7 +304,7 @@ async def test_tls(ops_test: OpsTest, kafka_app_charm):
     tls_config = {"generate-self-signed-certificates": "true", "ca-common-name": "kafka"}
 
     await asyncio.gather(
-        ops_test.model.deploy(TLS_NAME, channel="beta", config=tls_config, series="jammy"),
+        ops_test.model.deploy(TLS_NAME, channel="stable", config=tls_config, series="jammy"),
```

Comment:
Maybe we could change and use the [self-signed-opertor](https://charmhub.io/self-signed-certificates). The latter only works on juju 3.x and it should be a drop-in replacement for the tls-certificates-operator charm. 

I would give it a go, see [here](https://github.com/canonical/kafka-operator/commit/2b2bfaa17ff19e4d154ebebdfcb2b74aa2402e14) how it was integrated on kafka. If it does not work straight away, let's now waste time (since it is outside the scope of the PR) and we could just file an issue in the repo,  staying with what you have here.

Reply:
Done in 1ec40b4caeb653839bcce02483b562745ded6a8b